test/src/e2e_vm_tests/test_programs/should_pass/language/main_args/main_args_predicate/test.toml

```diff
@@ -1,4 +1,4 @@
-category = "run"
+category = "disabled"
```

Comment:
This test is disabled because it uses script data even though it's a predicate. That was passing because our E2E test config makes no distinction between predicates and scripts, and so the predicate was run a script. This now falls apart when predicates have main args.

Reply:
So do we need to update the harness to make that distinction and re-enable this test?

Reply:
Yeah exactly. Doing this in the SDK is much easier. In E2E tests, we'd have to manually craft a transaction with an input predicted by the predicate.